frontend/packages/console-dynamic-plugin-sdk/src/api/k8s-types.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  K8sResourceCommon,
+  CustomResourceDefinitionKind,
+  Selector,
+  OwnerReference,
+  Options,
+  Patch,
+} from '../extensions/console-types';
+import { K8sKind, ExtensionK8sGroupModel } from './common-types';
+
+// TODO - Convert WS factory to typescript and remove this temporary type
+type WSFactory = (id: any, options: any) => void;
+
+/**
+ * K8s CRUD utility types to be exposed by dynamic plugin SDK.
```

Comment:
Nit: I think it's better to represent this as non-block comment via `// ..` syntax.